dask/dataframe/tests/test_multi.py

```diff
@@ -308,6 +308,32 @@ def test_merge_asof_on_basic():
     assert_eq(c, C)
 
 
+def test_merge_asof_on_lefton_righton_error():
+    A = pd.DataFrame({"a": [1, 5, 10], "left_val": ["a", "b", "c"]})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({"a": [1, 2, 3, 6, 7], "right_val": [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    with pytest.raises(ValueError, match="combination of both"):
+        dd.merge_asof(a, b, on="a", left_on="left_val")
+        dd.merge_asof(a, b, on="a", right_on="right_val")
+        dd.merge_asof(a, b, on="a", left_on="left_val", right_on="right_val")
```

Comment:
We want to put each of these `dd.merge_asof(...)` calls inside their own `with pytest.raises(...):` context manager so that we ensure each `dd.merge_asof(...)` call raises the expected error. As is, we're only testing that one of those `dd.merge_asof(...)` calls raises an error